doc/source/serve/production.md

```diff
@@ -567,3 +557,216 @@ This section summarizes the best practices when deploying to production:
+
+In the `FruitStand` example above, let's change the price of a mango in the Serve config to 4:
+
+```console
+  - name: MangoStand
+    numReplicas: 1
+    userConfig: |
+      price: 4
+```
+
+Now to update the application we apply the modified manifest:
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+$ kubectl describe
+...
+  serveDeploymentStatuses:
+  - healthLastUpdateTime: "2022-07-18T21:51:37Z"
+    lastUpdateTime: "2022-07-18T21:51:41Z"
+    name: MangoStand
+    status: UPDATING
+...
+```
+
+If we query the application, we can see that we now get a different result reflecting the updated price:
+
+```console
+$ curl -X POST -H 'Content-Type: application/json' localhost:8000 -d '["MANGO", 2]'
+8
+```
+
+#### Updating the RayCluster config
+
+The process of updating the RayCluster config is the same as updating the Serve config.
+For example, we can update the number of worker nodes to 2 in the manifest:
+
+```console
+workerGroupSpecs:
+  # the pod replicas in this group typed worker
+  - replicas: 2
+```
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+$ kubectl describe
+...
+  pendingServiceStatus:
+    appStatus: {}
+    dashboardStatus:
+      healthLastUpdateTime: "2022-07-18T21:54:53Z"
+      lastUpdateTime: "2022-07-18T21:54:54Z"
+    rayClusterName: rayservice-sample-raycluster-bshfr
+    rayClusterStatus: {}
+...
+```
+
+In the status, you can see that the `RayService` is preparing a pending cluster.
+After the pending cluster is healthy, it will be switched to the active cluster and the previous cluster will be terminated.
+
+## Failure Recovery
+
+Ray Serve is resilient to any component failures within the Ray cluster out of the box.
+You can read more about how process and worker node failures are handled in: {ref}`serve-ft-detail`.
+
+### Head node failures
+
+By default the Ray head node is a single point of failure: if it crashes, the entire cluster crashes and needs to be restarted.
+When running on Kubernetes, the `RayService` controller will health check the cluster and restart it if this occurs, but it will still result in some downtime.
+In Ray 2.0, `KubeRay` has added experimental support for [GCS fault tolerance](https://ray-project.github.io/kuberay/guidance/gcs-ha/#ray-gcs-ha-experimental), preventing the Ray cluster from crashing if the head node goes down.
+While the head node is recovering, Serve applications can still handle traffic but will not be able to perform any updates or recover from other failures (e.g., actors or worker nodes crashing).
+Once the GCS is recovered, the cluster should return to normal behavior.
+
+### Health checking
+
+By default, each actor making up a Serve deployment is health checked and restarted on failure.
+
+:::{note}
+User-defined health checks are experimental and may be subject to change before the interface is stabilized. If you have any feedback or run into any issues or unexpected behaviors, please file an issue on GitHub.
```

Comment:
Can we link to the issue page on GitHub?

Reply:
oh yeah actually this is no longer experimental so i'll remove the note :)